File: app/src/main/java/com/pixelz360/docsign/imagetopdf/creator/RoomDb/AppDatabase.java
package com.pixelz360.docsign.imagetopdf.creator.RoomDb;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

@Database(entities = {PdfFile.class}, version = 2) // Increment version to 2
public abstract class AppDatabase extends RoomDatabase {

    public abstract PdfFileDao pdfFileDao();

    // Migration from version 1 to 2
    public static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(SupportSQLiteDatabase database) {
            // Example: Adding a new column "description"
            database.execSQL("ALTER TABLE PdfFile ADD COLUMN description TEXT DEFAULT ''");
        }
    };
}

File: app/src/main/java/com/pixelz360/docsign/imagetopdf/creator/PdfPythonEditorTools/AppDatabase.kt
package com.pixelz360.docsign.imagetopdf.creator.PdfPythonEditorTools

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase

@Database(entities = [PdfFile::class], version = 2)
abstract class AppDatabase : RoomDatabase() {
    abstract fun pdfFileDao(): PdfFileDao

    companion object {
        val MIGRATION_1_2 = object : Migration(1, 2) {
            override fun migrate(database: SupportSQLiteDatabase) {
                // Example: Add a column named "description" to PdfFile table
                database.execSQL("ALTER TABLE PdfFile ADD COLUMN description TEXT DEFAULT ''")
            }
        }
    }
}
